src/app-router.js

```diff
@@ -172,3 +150,53 @@ function targetIsThisWindow(target) {
     targetWindow === '_self' ||
     (targetWindow === 'top' && window === window.top);
 }
+
+function processResult(instruction, result, instructionCount, router) {
+  if (!(result && 'completed' in result && 'output' in result)) {
+    resut = result || {};
+    result.output = new Error(`Expected router pipeline to return a navigation result, but got [${JSON.stringify(result)}] instead.`);
+  }
+
+  let finalResult = null;
+  if (isNavigationCommand(result.output)) {
+    result.output.navigate(router);
+  } else {
+    finalResult = result;
+
+    if (!result.completed) {
+      let previousFragment = router.history.previousFragment;
+      if (previousFragment) {
+        router.navigate(router.history.previousFragment || '', false);
+      } else {
+        console.error('Router navigation failed, and no previous location could be restored.', result.output);
+      }
+    }
+  }
+
+  return router.dequeueInstruction(instructionCount + 1)
+    .then(innerResult => finalResult || innerResult || result);
+}
+
+function resolveInstruction(instruction, result, isInnerInstruction, router) {
+  instruction.resolve(result);
+
+  if (!isInnerInstruction) {
+    router.isNavigating = false;
+    let eventArgs = { instruction, result };
+    let eventName;
+
+    if (result.output instanceof Error) {
+      eventName = 'error';
+    } else if (!result.completed) {
+      eventName = 'canceled';
```

Comment:
I changed the inflection here to the American English version because it seems more common in other frameworks and libraries.